### PR TITLE
Set up associations between classes and objects.

### PR DESCRIPTION
**Pull request by milen-ale**
In this PR I managed to do the following

- [ ] Set up associations between classes and objects.
  - Created a class `Classroom` with the following:
  - `@label` instance variable
  - Setter and getter for `@label`
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student` and implemented the following
  - Created the `has-many` side (a classroom has many students).
  - Created the `belongs-to` side (a student belongs to a classroom).
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables
  - Setters and getters for instance variables 
- Created a class `Rental` with the following:
  - `@date` instance variable
  - Setter and getter for `@date` 
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: School/book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader rental
```
Comment (victorokoroji) [inline School/book.rb]: - [ ] KIndly re-write the attr_accessor properly by adding a colon before the rental attribute 👍🏽 

e.g. `attr_reader :rental`

File: School/classroom.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  attr_accessor label
```
Comment (victorokoroji) [inline School/classroom.rb]: - [ ] KIndly re-write the attr_accessor properly by adding a colon before the label attribute 👍🏽 

e.g. `attr_reader :label`

File: School/classroom.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader students
+
+  def initialize(label)
+    @label = label
+    @students = students[] # classroom has many students
```
Comment (victorokoroji) [inline School/classroom.rb]: - [ ] It is best you initialize the @students variable with just an empty array 👍🏽 

e.g `@students = []`

File: School/book.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initalize(title, author)
+    @title = title
+    @author = author
+    @rental = rental[]
```
Comment (victorokoroji) [inline School/book.rb]: - [ ] It is best you initialize the @rental variable with just an empty array 👍🏽

e.g @rental = []

File: School/book.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  def add_rental(rental)
```
Comment (victorokoroji) [inline School/book.rb]: - [ ] Kindly note that the book itself and the person should be sent as a parameter to create the new rental achieving the 'has-many' association. 
e.g.
  
```
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @milen-ale, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


## Required Changes ♻️

- [ ] Good job so far 👍🏽 . Please add the relationships/ associations for the `Person `class (person.rb) as well

_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: School/book.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    @rentals.push(rental)
```
Comment (t-yanick) [inline School/book.rb]: - [ ] As requested by the previous code reviewer,  book itself and the person should be sent as a parameter to create the new rental achieving the 'has-many' association. Your `add_rental` method is still missing person and book parameters. Please kindly fix that.

File: School/classroom.rb (hunk 2 above)
Comment (t-yanick) [inline School/classroom.rb]: - [ ] As requested by the previous code reviewer, please kindly correct the syntax error, re-write the attr_accessor properly by adding a `colon` before the label instance variable.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @milen-ale, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**